### PR TITLE
Add program status to the mental health status section of the patient…

### PR DESCRIPTION
**Pull request by louidorjp**
Screenshot from 2023-06-20 09-53-48
Please @mogoodrich @mseaton @dmdesimone review this PR.

### CODE REVIEW

File: configuration/pih/statusData/mentalHealth/lastDisposition.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+FROM obs o
+join encounter e on e.encounter_id = o.encounter_id
+where o.person_id = @patientId
+and o.concept_id = @dispositionCategory
+and o.voided = 0
+order by o.obs_datetime desc
+limit 1;
+
+select
+    @valueCoded as disposition;
```
Comment (mseaton) [inline configuration/pih/statusData/mentalHealth/lastDisposition.sql:17]: Defer to @dmdesimone , but we likely can use the existing function library that has been developed to make this entire sql statement easier.  All we likely need in this file is something like:

```
select value_coded as disposition from obs where obs_id = latest_obs(@patientId, 'PIH', '8620');
```

File: configuration/pih/statusData/mentalHealth/mentalHealthStatuses.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# These status data definitions are what define the HIV STATUS widget on the HIV Program Dashboard
+
+- id: "lastDisposition"
+  labelCode: "zl.status.lastDisposition"
+  statusDataQuery: "lastDisposition.sql"
+  valueExpression: "#if(!$disposition) $fn.translate('pihcore.none.label') #{else} $fn.translate($fn.format($fn.concept($disposition).getName())) #end"
```
[the remark this answers is not in the document]
Comment (louidorjp) [reply configuration/pih/statusData/mentalHealth/mentalHealthStatuses.yml:6]: Without the ".getName()", the concept name remains in English only, even if the language changes.

[the remark this answers is not in the document]
Comment (mogoodrich) [reply configuration/pih/statusData/mentalHealth/mentalHealthStatuses.yml:6]: @mseaton  @louidorjp it looks like the StatusDateFunctions format method (https://github.com/PIH/openmrs-module-pihcore/blob/master/api/src/main/java/org/openmrs/module/pihcore/status/StatusDataFunctions.java#L86) doesn't pass in the current locale when calling ObjectUtil.format() method, which in turn sets the locale to null and would result in the behavior (always being in English) @louidorjp  is seeing withou the "getName()"? 

However, in this code I don't think the "translate" call is needed (or does anything?)

**Review by mogoodrich — APPROVED** [review]
Added a comment about the name formatting, but otherwise looks good to me, thanks @louidorjp !

**Review by dmdesimone — COMMENTED** [review]
The SQL looks good to me.  Defer to others on the rest of the configuration.

**Pull request merged**